### DISCUSSION TITLE
Cypress setup for remote application monitoring

### DISCUSSION THREAD

**Discussion by m-vision** [cypress-io/cypress#28621] [Questions and Help]
Hello Cypress Community,

I am looking for the minimal local setup for testing several remote applications and workflows, where I don't have access to the code and not included in the CI/CD pipeline.

My first use cases are testing response from: url1, url2, url3 ,then...  login and do some specific workflow in each URL.

npm install cypress 

And anything else I need to install, configure, then start ?

Thanks!
Martin

**Comment by portyu9** [comment]
Hi @m-vision 

Yes. You can keep Cypress in a completely separate local project; you do **not** need access to the deployed application's source repository.

A minimal setup is:

```bash
npm install cypress --save-dev
npx cypress open
```

Choose **E2E Testing** and create your specs.

For your examples I would separate two kinds of checks:

**Endpoint/health checks**

```js
cy.request('https://example.com/url1')
  .its('status')
  .should('eq', 200)
```

**Browser workflows**

```js
cy.visit('https://example.com')
cy.get('[name="username"]').type('...')
// continue the actual workflow
```

The machine running Cypress only needs network access to those remote applications plus whatever credentials/configuration the tests require.

If one test navigates between different origins, current Cypress requires `cy.origin()` for commands executed on the additional origin. If the applications are independent, keeping them in separate tests/specs is usually simpler.

For repeated monitoring, run the Cypress project from whatever scheduler or CI system you control; no installation is required in the remote application's own repository.

Please mark this answer as accepted if it helped, thank you.
